bug: Filsystem.readdir() crashes the app with FilesystemDirectory.ExternalStorage on Android 10

# Bug Report

With Capacitor 2, calling `Filsystem.readdir()` with `FilesystemDirectory.Documents` or `FilesystemDirectory.ExternalStorage` system directories crashes the app (when run on Andoird 10).

## Capacitor Version

`npx cap doctor` output:

Latest Dependencies:
  @capacitor/cli: 2.1.0
  @capacitor/core: 2.1.0
  @capacitor/android: 2.1.0
  @capacitor/electron: 2.1.0
  @capacitor/ios: 2.1.0

Installed Dependencies:
  @capacitor/ios not installed
  @capacitor/electron not installed
  @capacitor/cli 2.1.0
  @capacitor/android 2.1.0
  @capacitor/core 2.1.0

[success] Android looking great!

## Affected Platform(s)

- [x] Android
- [ ] iOS
- [ ] Electron
- [ ] Web

## Current Behavior
Running the following code on Android 10 crashes the app:

```
Filsystem.readdir({
    directory: FilesystemDirectory.ExternalStorage,
    path: '/',
});
```

## Expected Behavior
Return an array of strings and keep running ;)

## Sample Code or Sample Application Repo

https://github.com/sebastienguillon/capacitor2-filesystem-bug

## Reproduction Steps

Clone and run the demo repo above on an Android 10 **device** or **demulator** and click on the last link (EXTERNAL_STORAGE). The app will crash (you will see a permissions popup first on the first run).

## Other Technical Details

`npm --version` output:
6.14.5

`node --version` output:
v12.16.1

## Other Information

Android Studio output at time of crash

> 2020-05-18 20:41:59.354 16910-16910/com.sebastienguillon.cap2fs I/Surface: opservice is null false
> 2020-05-18 20:41:59.373 16910-17042/com.sebastienguillon.cap2fs V/Capacitor/Plugin: To native (Capacitor plugin): callbackId: 69910929, pluginId: Filesystem, methodName: readdir
> 2020-05-18 20:41:59.374 16910-17042/com.sebastienguillon.cap2fs V/Capacitor: callback: 69910929, pluginId: Filesystem, methodName: readdir, methodData: {"directory":"EXTERNAL_STORAGE","path":"\/"}
> 2020-05-18 20:41:59.379 16910-17014/com.sebastienguillon.cap2fs V/Capacitor/Filesystem: Permission 'android.permission.READ_EXTERNAL_STORAGE' is granted
> 2020-05-18 20:41:59.381 16910-17014/com.sebastienguillon.cap2fs E/Capacitor: Serious error executing plugin
>     java.lang.reflect.InvocationTargetException
>         at java.lang.reflect.Method.invoke(Native Method)
>         at com.getcapacitor.PluginHandle.invoke(PluginHandle.java:99)
>         at com.getcapacitor.Bridge$1.run(Bridge.java:515)
>         at android.os.Handler.handleCallback(Handler.java:883)
>         at android.os.Handler.dispatchMessage(Handler.java:100)
>         at android.os.Looper.loop(Looper.java:214)
>         at android.os.HandlerThread.run(HandlerThread.java:67)
>      Caused by: java.lang.NullPointerException: Attempt to invoke virtual method 'java.lang.Class java.lang.Object.getClass()' on a null object reference
>         at org.json.JSONArray.<init>(JSONArray.java:117)
>         at com.getcapacitor.JSArray.<init>(JSArray.java:25)
>         at com.getcapacitor.JSArray.from(JSArray.java:48)
>         at com.getcapacitor.plugin.Filesystem.readdir(Filesystem.java:410)
>         at java.lang.reflect.Method.invoke(Native Method) 
>         at com.getcapacitor.PluginHandle.invoke(PluginHandle.java:99) 
>         at com.getcapacitor.Bridge$1.run(Bridge.java:515) 
>         at android.os.Handler.handleCallback(Handler.java:883) 
>         at android.os.Handler.dispatchMessage(Handler.java:100) 
>         at android.os.Looper.loop(Looper.java:214) 
>         at android.os.HandlerThread.run(HandlerThread.java:67) 
> 
> 
>     --------- beginning of crash
> 2020-05-18 20:41:59.383 16910-17014/com.sebastienguillon.cap2fs E/AndroidRuntime: FATAL EXCEPTION: CapacitorPlugins
>     Process: com.sebastienguillon.cap2fs, PID: 16910
>     java.lang.RuntimeException: java.lang.reflect.InvocationTargetException
>         at com.getcapacitor.Bridge$1.run(Bridge.java:524)
>         at android.os.Handler.handleCallback(Handler.java:883)
>         at android.os.Handler.dispatchMessage(Handler.java:100)
>         at android.os.Looper.loop(Looper.java:214)
>         at android.os.HandlerThread.run(HandlerThread.java:67)
>      Caused by: java.lang.reflect.InvocationTargetException
>         at java.lang.reflect.Method.invoke(Native Method)
>         at com.getcapacitor.PluginHandle.invoke(PluginHandle.java:99)
>         at com.getcapacitor.Bridge$1.run(Bridge.java:515)
>         at android.os.Handler.handleCallback(Handler.java:883) 
>         at android.os.Handler.dispatchMessage(Handler.java:100) 
>         at android.os.Looper.loop(Looper.java:214) 
>         at android.os.HandlerThread.run(HandlerThread.java:67) 
>      Caused by: java.lang.NullPointerException: Attempt to invoke virtual method 'java.lang.Class java.lang.Object.getClass()' on a null object reference
>         at org.json.JSONArray.<init>(JSONArray.java:117)
>         at com.getcapacitor.JSArray.<init>(JSArray.java:25)
>         at com.getcapacitor.JSArray.from(JSArray.java:48)
>         at com.getcapacitor.plugin.Filesystem.readdir(Filesystem.java:410)
>         at java.lang.reflect.Method.invoke(Native Method) 
>         at com.getcapacitor.PluginHandle.invoke(PluginHandle.java:99) 
>         at com.getcapacitor.Bridge$1.run(Bridge.java:515) 
>         at android.os.Handler.handleCallback(Handler.java:883) 
>         at android.os.Handler.dispatchMessage(Handler.java:100) 
>         at android.os.Looper.loop(Looper.java:214) 
>         at android.os.HandlerThread.run(HandlerThread.java:67) 
> 
> 
>     --------- beginning of system
> 2020-05-18 20:41:59.429 16910-17014/com.sebastienguillon.cap2fs I/Process: Sending signal. PID: 16910 SIG: 9

### Notes

The same code (see demo repo above) will not crash the app on Android versions below 10 (tested on real devices and emulators).
Also the crash does not happen when creating the exact same project with capacitor 1.5.2.

--- a/android/capacitor/src/main/java/com/getcapacitor/plugin/Filesystem.java
+++ b/android/capacitor/src/main/java/com/getcapacitor/plugin/Filesystem.java
@@ -412,10 +412,13 @@ public class Filesystem extends Plugin {
          || isStoragePermissionGranted(PluginRequestCodes.FILESYSTEM_REQUEST_READ_FOLDER_PERMISSIONS, Manifest.permission.READ_EXTERNAL_STORAGE)) {
       if (fileObject != null && fileObject.exists()) {
         String[] files = fileObject.list();
-
-        JSObject ret = new JSObject();
-        ret.put("files", JSArray.from(files));
-        call.success(ret);
+        if (files != null) {
+          JSObject ret = new JSObject();
+          ret.put("files", JSArray.from(files));
+          call.success(ret);
+        } else {
+          call.error("Unable to read directory");
+        }
       } else {
       call.error("Directory does not exist");
       }

